src/cats/monad/channel.cljc

```diff
@@ -49,19 +40,13 @@
 (def ^{:no-doc true}
   channel-monad
   (reify
-    proto/Functor
-    (fmap [mn f mv]
-      (let [ctx m/*context*
-            channel (chan)]
-        (take! mv (fn [v]
-                    (put! channel
-                          ;; Set double monad for handle properly
-                          ;; monad transformers
-                          (m/with-monad ctx
```

Comment:
This with-monad is required, because the take operation can be asynchronous and can be executed in another thread where the context is not set.